storage/src/main/java/org/apache/uniffle/storage/handler/impl/MemoryQuorumClientReadHandler.java

```diff
@@ -65,11 +72,26 @@ public ShuffleDataResult readShuffleData() {
       }
     }
 
-    if (!readSuccessful) {
+    if (!readSuccessful && !readAfterFallback) {
```

Comment:
Why do we throw an exception here?

Reply:
Because we don't know where the missing blocks be, even if read fail from memory after fallback, these blocks may also be complete.